File: src/main/java/insogroup/utils/utils/IOC/interfaces/IOCStrategy.java
package insogroup.utils.utils.IOC.interfaces;

import insogroup.utils.utils.IOC.exceptions.ResolveStrategyException;

/**
 * Using by IOC for getting instance from strategy
 */
public interface IOCStrategy {
    /**
     * Using by IOC for getting instance from strategy
     * @param args Args for using in strategy
     * @return new, cached or other instance of object
     * @throws ResolveStrategyException Throw when instance can't be resolved
     */
    Object getInstance(Object... args) throws ResolveStrategyException;
}

File: src/main/kotlin/com/insogroup/utils/IOC/interfaces/IOCStrategy.kt
package com.insogroup.utils.IOC.interfaces

import com.insogroup.utils.IOC.exceptions.ResolveStrategyException

/**
 * Using by IOC for getting instance from strategy
 */
interface IOCStrategy {
    /**
     * Using by IOC for getting instance from strategy
     * @param args Args for using in strategy
     * *
     * @return new, cached or other instance of object
     * *
     * @throws ResolveStrategyException Throw when instance can't be resolved
     */
    @Throws(ResolveStrategyException::class)
    fun getInstance(vararg args: Any): Any
}
